### PR TITLE
Add Sphinx documentation for the Python bindings

### LINKED ISSUE #154: Create sphinx doc for Python binding
At the very minimum, we should have a doc that list all the available APIs in the python binding

### PR DESCRIPTION
**Pull request by fvaleye**
# Description
The Sphinx documentation is applied to the docstrings used in the Python code.
The typing of Python is also used for detecting the types of parameters inside the functions.

# Related Issue(s)
- closes #154 

# Documentation
We could link Github with the Read the Docs page but I don't have the permissions 😃 

> Quickstart for GitHub-Hosted Projects
> By the end of this quickstart, you will have a new project automatically updated when you push to GitHub.
> 
> Create an account on Read the Docs. You will get an email verifying your email address which you should accept within 7 days.
> Log in and click on "Import a Project".
> Click "Connect to GitHub" in order to connect your account's repositories to GitHub.
> When prompted on GitHub, give access to your account.
> Click "Import a Repository" and select any desired repository.
> Change any information if desired and click "Next".
> All done. Commit away and your project will auto-update.
>

### CODE REVIEW

**Comment by houqp** [comment]
@fvaleye could you add a doc build step in CI to help guard against doc breaking changes in the future?

**Comment by fvaleye** [comment]
> @fvaleye could you add a doc build step in CI to help guard against doc breaking changes in the future?

Yes, you are right 👍 
I added a check documentation step for the CI.

**Review by rtyler — APPROVED** [review]

File: python/docs/source/conf.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    """
+    cargo_content = toml.load("../../Cargo.toml")
+    return cargo_content["package"]["version"]
+
+
+# -- Project information -----------------------------------------------------
+
+project = "delta-rs"
+copyright = ""
+author = "Florian Valeye"
```
Comment (fvaleye) [inline python/docs/source/conf.py:34]: @houqp we may need to change this too before merging

File: python/README.md [hunk 2]
```diff
@@ -19,65 +19,9 @@ objection store communication. Please file Github issue to request for critical
...
-
-```
->>> from deltalake import DeltaTable
->>> dt = DeltaTable("../rust/tests/data/simple_table")
->>> dt.schema()
-Schema(Field(id: DataType(long) nullable(True) metadata({})))
->>> dt.pyarrow_schema()
-id: int64
-```
+<!--- Add the readthedocs link --->
```
[the remark this answers is not in the document]
Comment (fvaleye) [reply python/README.md:24]: Yes with the copyright and author sections above.

File: python/docs/source/conf.py (hunk 1 above)
Comment (houqp) [reply python/docs/source/conf.py:34]: @rtyler do you have any preference on this?

File: python/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (houqp) [reply python/README.md:24]: @rtyler are you cool with us adding readthedocs integration? or do you prefer to self-host with github pages instead?

File: python/docs/source/conf.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
```
Comment (rtyler) [inline python/docs/source/conf.py:34]:
```suggestion
copyright = "2021 delta-rs contributors"
author = "delta-rs contributors"
```

I think it's best to leave this general and vague, not that the Sphinx configuration has any legal basis anyways :smile_cat:

**Comment by houqp** [comment]
I don't have the access to turn on readthedocs integration, so I pushed a local build to gh-pages branch. @fvaleye could you take a look at https://github.com/fvaleye/delta-rs/pull/4?

**Comment by houqp** [comment]
We can work on the docs release automation as a follow up task.

**Pull request merged**